tools/bin/load_test/load_test_airbyte.sh

```diff
@@ -0,0 +1,252 @@
+      ;;
+    *)
+      showhelp
+      ;;
+  esac
+done
+
+function setup {
+  set -e
+  if test -z "$workspace_id"; then
+    echo "error: must set a workspace id with -W"
+    exit 1
+  fi
+
+  echo "set workspace_id to              ${workspace_id}"
+  echo "set hostname to                  ${hostname}"
+  echo "set api_port to                  ${api_port}"
+  echo "set x_endpoint_header to         ${x_endpoint_header}"
+  echo "set num_connections to           ${num_connections}"
+  echo "set sync_minutes to              ${sync_minutes}"
+
+  setCleanupFilesForWorkspace $workspace_id
+
+  mkdir -p cleanup
+
+  touch $connection_cleanup_file
+  touch $source_cleanup_file
+  touch $destination_cleanup_file
+}
+
+function getE2ETestSourceDefinitionId {
+  # call source_definitions/list and search response for the E2E Test dockerRepository to get the ID.
+  # local uses `source-e2e-test`, while cloud uses `source-e2e-test-cloud`
+  sourceDefinitionId=$(
+    callApi "source_definitions/list" |
+      jq -r '.sourceDefinitions[] |
+        select(
+          (.dockerRepository == "airbyte/source-e2e-test") or
+          (.dockerRepository == "airbyte/source-e2e-test-cloud")
+        ) |
+        .sourceDefinitionId'
+  )
+  export sourceDefinitionId
+}
+
+function getE2ETestDestinationDefinition {
+  # call destination_definitions/list and search response for the E2E Test dockerRepository to get the ID.
+  # local uses `destination-dev-null`, while cloud uses `destination-e2e-test-cloud`
+  destinationDefinitionId=$(
+    callApi "destination_definitions/list" |
+      jq -r '.destinationDefinitions[] |
+        select(
+          (.dockerRepository == "airbyte/destination-e2e-test") or
+          (.dockerRepository == "airbyte/destination-dev-null")
+        ) |
+        .destinationDefinitionId'
+  )
+  export destinationDefinitionId
+}
+
+function createSource {
+  body=$(
+    sed "
+      s/replace_source_read_secs/$(( 60*sync_minutes ))/g ;
+      s/replace_source_definition_id/$sourceDefinitionId/g ;
+      s/replace_workspace_id/$workspace_id/g" source_spec.json |
+    tr -d '\n' |
+    tr -d ' '
+  )
+
+  sourceId=$(
+    callApi "sources/create" $body |
+    jq -r '.sourceId'
+  )
+  export sourceId
+  echo $sourceId >> $source_cleanup_file
+}
+
+function createDestination {
+  body=$(
+    sed "
+      s/replace_destination_definition_id/$destinationDefinitionId/g ;
+      s/replace_workspace_id/$workspace_id/g" destination_spec.json |
+    tr -d '\n' |
+    tr -d ' '
+  )
+  destinationId=$(
+    callApi "destinations/create" $body |
+    jq -r '.destinationId'
+  )
+  export destinationId
+  echo $destinationId >> $destination_cleanup_file
+}
+
+function discoverSource {
+  sourceCatalogId=$(
+    callApi "sources/discover_schema" "{\"sourceId\":\"$sourceId\",\"disable_cache\":true}" |
+      jq -r '.catalogId'
+  )
+  export sourceCatalogId
+}
+
+function createMultipleConnections {
+  for i in $(seq 1 $num_connections)
```

Comment:
is this 100 items or 100000 items?

if there is a real mystery to the user about how long this will take I like printing the total number so the user can estimate time to completion

Reply:
Good suggestion, added an `out of X connections` to the echo